src/components/Users/UserAvailabilityTab.tsx

```diff
@@ -85,6 +85,17 @@ export default function UserAvailabilityTab({ userData: user }: Props) {
     }),
   });
 
+  const schedulableUsersQuery = useQuery({
+    queryKey: ["practitioners", facilityId],
+    queryFn: query(scheduleApis.appointments.availableUsers, {
+      pathParams: { facility_id: facilityId },
+    }),
+  });
+
+  const availableUserId = schedulableUsersQuery.data?.users.find(
+    (f) => f.id == user.id,
+  );
```

Comment:
this can be simplified further by leveraging `select` option of `useQuery` and by switching to using `Array.some` instead of `Array.find`

```suggestion
  const { data: isSchedulableResource } = useQuery({
    queryKey: ["practitioners", facilityId],
    queryFn: query(scheduleApis.appointments.availableUsers, {
      pathParams: { facility_id: facilityId },
    }),
    select: (data) => data.users.some(({ id }) => user.id === id ),
  });
```

Reply:
React Query treats data as unknown
so can we manually cast data Inside select like this
select: (data) => (data as { users: UserBase[] }).users.some(({ id }) => user.id === id),

Reply:
Yes!

Reply:
you can do this:

```
select: (data: { users: UserBase[] }) => data.users.some(({ id }) => user.id === id)
```